### PR TITLE
[jax2tf] Remove legacy bfloat16 conversion code.

### PR DESCRIPTION
**Pull request by bchetioui**
Now that `tf.bfloat16` and `jnp.bfloat16` are the same, there should not be a need for specific conversion code anymore between the two.

### CODE REVIEW

**Comment by bchetioui** [comment]
@googlebot I signed it!

File: jax/lib/__init__.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 import jaxlib
 
 # Must be kept in sync with the jaxlib version in build/test-requirements.txt
-_minimum_jaxlib_version = (0, 1, 55)
+_minimum_jaxlib_version = (0, 1, 58)
```
Comment (gnecula) [inline jax/lib/__init__.py:26]: Do we really need to change this, or is it enough to use a new version of tf-nightly? In other words, does the jnp.bfloat16 from jaxlib 0.1.55 match the tf.bfloat16 from tf-nightly?

**Review by gnecula — COMMENTED** [review]
Thanks

File: jax/lib/__init__.py (hunk 1 above)
Comment (bchetioui) [reply jax/lib/__init__.py]: No, you're right. I thought I recalled we mentioned this was needed before, but it seems it is not. Thanks!

Comment (gnecula) [reply jax/lib/__init__.py:26]: There have been some problems with 0.1.58 (#5374). I am reluctant to force people to upgrade at the moment.

**Comment by gnecula** [comment]
I am proposing that we hold off on merging this until jaxlib minimum version needs to be upgraded for other reasons as well.

**Comment by bchetioui** [comment]
Oops, I meant to answer sooner! Sounds good to me, thanks :+1:

File: jax/lib/__init__.py (hunk 1 above)
Comment (gnecula) [reply jax/lib/__init__.py:26]: I am going to be merging now, since minimum jaxlib was bumped to 1.59

**Pull request merged**